source/common/access_log/access_log_formatter.cc

```diff
@@ -280,6 +281,18 @@ std::vector<FormatterProviderPtr> AccessLogFormatParser::parse(const std::string
         parseCommand(token, start, ":", filter_namespace, path, max_length);
         formatters.emplace_back(
             FormatterProviderPtr{new DynamicMetadataFormatter(filter_namespace, path, max_length)});
+      } else if (absl::StartsWith(token, FILTER_STATE_TOKEN)) {
+        std::string key;
+        absl::optional<size_t> max_length;
+        std::vector<std::string> path;
+        const size_t start = FILTER_STATE_TOKEN.size();
+
+        parseCommand(token, start, ":", key, path, max_length);
+        if (!path.empty()) {
+          throw EnvoyException("Invalid filter state configuration. Key contains ':'.");
```

Comment:
This is kind of a strange condition anyway to be extra defensive about, can you elaborate?

Reply:
fixed both